core/trino-main/src/main/java/io/trino/execution/SqlTaskManager.java

```diff
@@ -656,6 +657,17 @@ public TaskInfo failTask(TaskId taskId, Throwable failure)
         return tasks.getUnchecked(taskId).failed(failure);
     }
 
+    public void cleanupTask(TaskId taskId)
+    {
+        requireNonNull(taskId, "taskId is null");
+        SqlTask sqlTask = tasks.getIfPresent(taskId);
```

Comment:
nit: Could there ever be a race condition, e.g: task was not created yet?

Reply:
Makes sense to make this idemptotent